src/Orleans.Runtime.Abstractions/Hosting/HostingGrainExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Orleans.CodeGeneration;
+using Orleans.Runtime;
+
+namespace Orleans.Hosting
+{
+    /// <summary>
+    /// Methods for configuring <see cref="IGrainExtension"/>s on a silo.
+    /// </summary>
+    public static class HostingGrainExtensions
+    {
+        /// <summary>
+        /// Registers a grain extension implementation for the specified interface.
+        /// </summary>
+        /// <typeparam name="TExtensionInterface">The <see cref="IGrainExtension"/> interface being registered.</typeparam>
+        /// <typeparam name="TExtension">The implementation of <typeparamref name="TExtensionInterface"/>.</typeparam>
+        public static ISiloHostBuilder AddGrainExtension<TExtensionInterface, TExtension>(this ISiloHostBuilder builder)
```

Comment:
Should we provide a way to filter in which grain types (and/or instances thinking about Orleankka) the extension is installed?

Reply:
I suspect there will be cases where users will want more control over which grains extensions are installed on.  We could probably do this via silo configuration, attribution, or some sort of requirements or validation check..  I suggest we expand this feature as needed, rather than speculate what users will need.   In any case, this core capability is a prerequisite to any more intelligent installation of extensions.